File: typescript/packages/bench/src/rules/technicianCapacityRule.ts
/** Cardinality fixture: nominal (class, `implements`) implementor of RuleContract. */

import type { RuleContract } from "../contracts/ruleContract";
import type { Dataset } from "../dataset";

export class TechnicianCapacityRule implements RuleContract {
    readonly key = "technician-capacity";

    evaluate(data: Dataset): boolean {
        return data.technicians.length >= 1;
    }
}

File: rust/crates/atelier-bench/src/rules/technician_capacity_rule.rs
//! Rule 5/24 (textual half).

use crate::contracts::rule_contract::RuleContract;
use crate::dataset::Dataset;

pub struct TechnicianCapacityRule;

impl RuleContract for TechnicianCapacityRule {
    fn evaluate(&self, data: &Dataset) -> bool {
        !data.technicians.is_empty()
    }
}
